lib/mini_mime.rb

```diff
@@ -60,18 +58,14 @@ def self.lookup_by_filename(filename)
     end
 
     def self.lookup_by_extension(extension)
-      LOCK.synchronize do
-        @db ||= new
-        @db.lookup_by_extension(extension) ||
-          @db.lookup_by_extension(extension.downcase)
-      end
+      @db ||= new
```

Comment:
There is a small race condition here, that could lead to more than one DB to be instantiated, but only one will be kept so not sure if it's a big deal.

We could eagerly instantiate the db, and reset it when the paths are changed, but that would change the semantic a bit.

Reply:
Looks OK to me, but I guess there could be a performance impact if a lot of threads try to init the DB at the same time (e.g. because they're all executing the same startup code in lockstep). A possible alternative to avoid multiple initialization could be the DCL-like idiom `@db || MUTEX.synchronize { @db ||= new }`.

Reply:
Yeah, that's a good idea. Not sure why I didn't think about it.

Reply:
@casperisfine PR here: https://github.com/discourse/mini_mime/pull/56

Reply:
Thanks, note that I'm not maintainer though, it's up to @SamSaffron to merge this or not.